estate/models/estate_property.py

```diff
@@ -0,0 +1,93 @@
+from odoo import api, fields, models
+from dateutil.relativedelta import relativedelta
+from datetime import datetime
+from odoo.exceptions import UserError
+
+
+class EstateProperty(models.Model):
+    _name = "estate_property"
+    _description = "Estate Property"
+    name = fields.Char("Title", required=True)
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date(
+        copy=False, default=datetime.now() + relativedelta(months=3)
+    )
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float(readonly=True, copy=False)
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer("Living Area (sqm)")
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer()
+    garden_orientation = fields.Selection(
+        string="Orientation",
+        selection=[
+            ("north", "North"),
+            ("south", "South"),
+            ("east", "East"),
+            ("west", "West"),
+        ],
+    )
+    active = fields.Boolean(default=True)
+    state = fields.Selection(
+        readonly=True,
+        string="State",
+        selection=[
+            ("new", "New"),
+            ("offer_received", "Offer Received"),
+            ("offer_accepted", "Offer Accepted"),
+            ("sold", "Sold"),
+            ("cancelled", "Cancelled"),
+        ],
+        default="new",
+        required=True,
+        copy=False,
+    )
+    type_id = fields.Many2one("estate_property_type", string="Property Type")
+    salesperson_id = fields.Many2one(
+        "res.users", string="Salesperson", default=lambda self: self.env.user
+    )
+    buyer_id = fields.Many2one("res.partner", string="Buyer", copy=False)
+    tag_ids = fields.Many2many("estate_property_tag")
+    offer_ids = fields.One2many("estate_property_offer", "property_id", string="Offers")
+    total_area = fields.Integer("Total Area (sqm)", compute="_compute_area")
+    best_price = fields.Float("Best Price", compute="_compute_best_price")
+
+    @api.depends("living_area", "garden_area")
+    def _compute_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends("offer_ids.price")
+    def _compute_best_price(self):
+        for record in self:
+            record.best_price = (
+                0
+                if len(record.offer_ids) == 0
+                else max(offer.price for offer in record.offer_ids)
+            )
```

Comment:
the `.mapped('field')` return a list with the all the field values for the recordset and the `or [0]` will replace the list of prices with list contain 0 if the list is empty list
```suggestion
            record.best_price = max(record.offer_ids.mapped('price') or [0])
```